connectors/apache-connector/src/main/java/org/glassfish/jersey/apache/connector/ApacheConnector.java

```diff
@@ -631,8 +619,13 @@ private static InputStream getInputStream(final CloseableHttpResponse response)
         return new FilterInputStream(inputStream) {
             @Override
             public void close() throws IOException {
-                response.close();
-                super.close();
+                try {
+                    super.close();
+                } catch (IOException ex) {
+                    // Ignore
+                } finally {
+                    response.close();
+                }
```

Comment:
Thanks @agherardi for fixing this

2 years down the line and because of missing `response.close()` in previous versions of Jersey, we finally figured out the root cause.

Our leased http connections never were returned to the pool because response wasn't being explicitly read into entities in some cases. 

Our request threads talked to a database and had written few rows (REPEATABLE_READ). As a result, the JDBI `handle` was never closed (thread was in PARKED state due to no more available connections in the pool) and exclusive locks were held as the request was neither rolled back nor committed but suspended.

We have finally figured out the root cause.

Reply:
Please understand that this also depends on the Apache HTTP Client version. See also a [followup](https://github.com/eclipse-ee4j/jersey/issues/4321) discussion.

Reply:
Thank you @jansupol. Following.